plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -78,16 +202,30 @@ export const EntityRefLink = forwardRef<any, EntityRefLinkProps>(
     );
 
     const link = (
-      <Link {...linkProps} ref={ref} to={entityRoute(routeParams)}>
+      <Link
+        {...bindHover(popupState)}
+        {...linkProps}
+        ref={ref}
+        to={entityRoute(routeParams)}
+      >
         {children}
         {!children && (title ?? formattedEntityRefTitle)}
       </Link>
     );
 
-    return title ? (
-      <Tooltip title={formattedEntityRefTitle}>{link}</Tooltip>
-    ) : (
-      link
+    return (
+      <>
+        {title ? (
+          <Tooltip title={formattedEntityRefTitle}>{link}</Tooltip>
+        ) : (
+          link
+        )}
+
+        <PeekAheadPopover
```

Comment:
OK so it'll have one popover instance per every single link in the render output, maybe no easy way to get around that, but it does weigh down the page rendering.

Reply:
Yeah its not great. Like if you have user:guest 10 times on the screen, it'll retrieve the thing from the catalog for each hover as well. I couldnt think of a nice way to improve that situation.